tests/conftest.py

```diff
@@ -236,6 +236,22 @@ async def watch_kubernetes(kube_client, kube_ns):
         raise exc
 
 
+async def _delete_namespace(client, namespace):
+    await client.delete_namespace(namespace, body={}, grace_period_seconds=0)
+    for _ in range(20):  # Usually finishes a good deal faster
+        try:
+            await client.read_namespace(namespace)
+        except ApiException as e:
+            if e.status == 404:
+                return
+            else:
+                raise
+        else:
+            print("waiting for %s to delete" % namespace)
+            await asyncio.sleep(1)
+    raise Exception(f"Namespace {namespace} not deleted after 20 s")
```

Comment:
This line is new. Previously this function would wait up to 20s but not do anything if the namespace hadn't been deleted. Is that the behaviour we desire?

Reply:
What you did now is better!